Makepad-fr/vif.io
# chore: use structured data for user profile

Closes #9 

## Added

- Data structures for user profile details, links, and metadata
- Service level function to retrieve user profile details and links
- Add new templates for 404, landing page and create yours pages 

## Changed

- Update some basic CSS for links
- Move handler and middleware code to the `internal` package
- 

## Fixed

- Checking if the user exists issue fixed

# How to test these changes?

- Start the server with the following command on branch `chore/profile-structure`:

```bash
DATABASE_URL="toto" DATABASE_USER="mysuperdatabaseuser" DATABASE_PASSWORD="helloworld" make start
```

- Go to `localhost:4321/kaanyagci` 
- Page should render  with no JavaScript and HTML errors
